src/mailroom.py

```diff
@@ -0,0 +1,196 @@
+# -*- coding: utf-8 -*-
+"""
+The mailroom script allows someone to add donors to a database and generate a
+thank you letter. It also allows the user to view the list of donors in a table
+format, sorted by total amount donated. The data is persisted between
+application runs.
+"""
+
+import os
+import sys
+import io
+
+DEFAULT_DONORS = {
+    'Steven': [3, 6],
+    'David': [1, 5, 7],
+    'Zach': [5, 3, 6]
+}
+
+DEFAULT_LETTER = 'Thank you, {name} for your donation of {amount}.\n'
+
+
+def main_menu():    # pragma: no cover
+    """Main func that get executed when run in the CLI."""
+    try:
+        while True:
+            clear_screen()
+            print(welcome_message())
+            user_input = input("Selection: ")
+            if user_input not in ['1', '2', '3']:
+                continue
+            handle_input(user_input)()
+    except KeyboardInterrupt:
+        exit()
+
+
+def send_thanks():  # pragma: no cover
+    """Send thank-you note submenu."""
+    while True:
+        donor = input("Enter Donor name, 'list' or 'return': ")
+        if donor == 'list':
+            clear_screen()
+            print(build_report_table(donor_list_by_total(donor_dict)))
+        elif donor == 'return':
+            clear_screen()
+            return
+        elif donor == '':
+            continue
+        else:
+            amount = 0
+            while not amount:
+                amount = validate_donation(
+                    input("Input donation amount: "))
+            else:
+                donor_dict.setdefault(donor, []).append(amount)
+            print('\n' + generate_thankyou(donor, format_amount(amount)))
+
+
+def report_donors():    # pragma: no cover
+    """Report submenu."""
+    clear_screen()
+    donor_list = donor_list_by_total(donor_dict)
+    print(build_report_table(donor_list))
+
+
+def report_donors_wait():    # pragma: no cover
+    """Report submenu."""
+    report_donors()
+    press_to_continue()
+
+
+def exit():     # pragma: no cover
+    """Terminate program without error."""
+    print(save_donors(donor_dict))
+    print('Script terminated.\n')
+    sys.exit(0)
```

Comment:
I like this encapsulation.  Places all the actions you want to take in-line with the act of quitting.  

However, what happens if there's an I/O error when you try to write out your donor file.  The fact that this code is un-tested is a potential weakness.  You could exit the script on an exception without having saved things at all.